### PR TITLE
UTF8 is no real UTF8 in MySQL, one needs to use utf8mb4

### PR DESCRIPTION
**Pull request by csware**
This allows to use emojis in vacation mails and allow characters in names.

The usage of UTF8mb4 should be the default or at least configurable. utf8mb4 was introuced in MySQL 5.5.3 IIRC.

### CODE REVIEW

**Comment by csware** [comment]
Also relates to #337.

**Comment by csware** [comment]
Of course at least the mailbox.name and vacation.subject and vatation.body columns need to be utf8mb4 (i use `utf8mb4_bin` as no caseinsensitive comparisons are needed).

**Comment by DavidGoodwin** [comment]
current vacation table looks like : 

```
CREATE TABLE `vacation` (
  `email` varchar(255) NOT NULL,
  `subject` varchar(255) NOT NULL,
  `body` text NOT NULL,
  `activefrom` timestamp NOT NULL DEFAULT '2000-01-01 00:00:00',
  `activeuntil` timestamp NOT NULL DEFAULT '2038-01-18 00:00:00',
  `cache` text NOT NULL,
  `domain` varchar(255) NOT NULL,
  `interval_time` int(11) NOT NULL DEFAULT 0,
  `created` datetime NOT NULL DEFAULT '2000-01-01 00:00:00',
  `modified` timestamp NOT NULL DEFAULT current_timestamp(),
  `active` tinyint(1) NOT NULL DEFAULT 1,
  PRIMARY KEY (`email`),
  KEY `email` (`email`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1 COMMENT='Postfix Admin - Virtual Vacation'
```

**Comment by DavidGoodwin** [comment]
alter table vacation change body body text charset utf8mb4 not null;

etc.

File: functions.inc.php [hunk 1]
```diff
@@ -1699,8 +1699,8 @@ function db_connect()
 
             $options[PDO::MYSQL_ATTR_SSL_VERIFY_SERVER_CERT] = (bool)$verify;
         }
-        $queries[] = 'SET CHARACTER SET utf8';
-        $queries[] = "SET COLLATION_CONNECTION='utf8_general_ci'";
+        $queries[] = 'SET CHARACTER SET utf8mb4';
```
[the remark this answers is not in the document]
Comment (csware) [reply functions.inc.php]: See second commit

File: VIRTUAL_VACATION/vacation.pl [hunk 2]
```diff
@@ -208,7 +208,7 @@
 
 my $db_true; # MySQL and PgSQL use different values for TRUE, and unicode support...
 if ($db_type eq 'mysql') {
-    $dbh->do('SET CHARACTER SET utf8;');
```
Comment (DavidGoodwin) [inline VIRTUAL_VACATION/vacation.pl:211]: Should perl be using SET NAMES ... here ?

Comment (csware) [reply VIRTUAL_VACATION/vacation.pl:211]: Possibly, works this way for me.

**Comment by csware** [comment]
This was quick! I had expected that mergig this would take longer ;)

Have a nice weekend!

**Pull request merged**